src/components/Carousel.tsx

```diff
@@ -1,18 +1,215 @@
-import React from 'react';
+import { Component } from 'react';
 import './Carousel.scss';
 
-const Carousel: React.FC = () => (
-  <div className="Carousel">
-    <ul className="Carousel__list">
-      <li><img src="./img/1.png" alt="1" /></li>
-      <li><img src="./img/1.png" alt="2" /></li>
-      <li><img src="./img/1.png" alt="3" /></li>
-      <li><img src="./img/1.png" alt="4" /></li>
-    </ul>
-
-    <button type="button">Prev</button>
-    <button type="button">Next</button>
-  </div>
-);
+type Props = {
+  images: string[];
+};
+
+type State = {
+  itemWidth: number;
+  frameSize: number;
+  step: number;
+  currentPosition: number
+  animationDuration: number;
+  disabledValue: boolean;
+  infinite: boolean;
+};
+
+class Carousel extends Component<Props, State> {
+  state: Readonly<State> = {
+    itemWidth: 130,
+    frameSize: 2,
+    step: 2,
+    currentPosition: 0,
+    animationDuration: 1000,
+    disabledValue: false,
+    infinite: false,
+  };
+
+  newPosition = 0;
+
+  moveLeft = () => {
+    const {
+      itemWidth,
+      step,
+      currentPosition,
+      frameSize,
+      infinite,
+    } = this.state;
+
+    this.newPosition = currentPosition + itemWidth * step;
+
+    if (this.newPosition > 0) {
+      this.newPosition = 0;
+
+      if (infinite) {
+        this.newPosition = -(
+          (this.props.images.length - frameSize) * itemWidth);
```

Comment:
such a record is not obvious, try to divide operations into variables

```suggestion
        // myVariable = this.props.images.length - frameSize
        this.newPosition = -(myVariable * itemWidth);
```